### DISCUSSION TITLE
Deployment Issues with React App ! Any help appreciated!

### DISCUSSION THREAD

**Discussion by Erez-Michael** [vercel/next.js#46162] [Help]
I get the following message when trying to deploy:

[12:51:04.739] Running build in Washington, D.C., USA (East) – iad1
[12:51:04.789] Cloning github.com/Erez-Michael/ludmer-engineering (Branch: main, Commit: 53516e6)
[12:51:05.080] Previous build cache not available
[12:51:08.526] Cloning completed: 3.736s
[12:51:08.700] Running "vercel build"
[12:51:09.144] Vercel CLI 28.16.2
[12:51:09.726] Installing dependencies...
[12:51:15.934] npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
[12:51:16.116] npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
[12:51:16.151] npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
[12:51:16.306] npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
[12:51:20.028] npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
[12:51:45.363] npm ERR! code 1
[12:51:45.364] npm ERR! path /vercel/path0/node_modules/@fortawesome/free-solid-svg-icons
[12:51:45.365] npm ERR! command failed
[12:51:45.365] npm ERR! command sh -c -- node attribution.js
[12:51:45.365] npm ERR! node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
[12:51:45.365] npm ERR! node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
[12:51:45.367] 
[12:51:45.367] npm ERR! A complete log of this run can be found in:
[12:51:45.367] npm ERR!     /vercel/.npm/_logs/2023-02-20T17_51_10_005Z-debug-0.log
[12:51:45.404] Error: Command "npm install" exited with 1
[12:51:45.476]

**Comment by icyJoseph** [comment]
And you don't have any special build commands right? I'd ask the Vercel support team for this one. It's like your Nodejs version is missing something, but it's like a standard environment for everyone...

**Comment by peterwb** [comment]
Any resolution on this, getting same issue here.

**Reply by petmarsh** [reply]
More or less looks like a pre-compiled LTS/Current version of nodejs running on an outdated system - the team bumped up the glibc requirements for v18:

_Prebuilt binaries for Linux are now built on Red Hat Enterprise Linux (RHEL) 8 and are compatible with Linux distributions based on glibc 2.28 or later, for example, Debian 10, RHEL 8, Ubuntu 20.04_

**Comment by GabrielIdongesit** [comment]
this is errors i got trying to deploy
how do I fix it please

16:04:22.228 Running build in Washington, D.C., USA (East) – iad1
16:04:22.228 Build machine configuration: 2 cores, 8 GB
16:04:22.404 Cloning github.com/GabrielIdongesit/emotionsx (Branch: main, Commit: ea77a77)
16:04:22.405 Previous build caches not available.
16:04:23.074 Cloning completed: 670.000ms
16:04:24.673 Running "vercel build"
16:04:25.112 Vercel CLI 50.1.3
16:04:25.662 Running "install" command: `npm install`...
16:04:29.552 npm warn deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
16:04:29.837 npm warn deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
16:04:29.921 npm warn deprecated workbox-cacheable-response@6.6.0: workbox-background-sync@6.6.0
16:04:30.026 npm warn deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
16:04:30.117 npm warn deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
16:04:30.119 npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
16:04:30.470 npm warn deprecated q@1.5.1: You or someone you depend on is using Q, the JavaScript Promise library that gave JavaScript developers strong feelings about promises. They can almost certainly migrate to the native JavaScript promise now. Thank you literally everyone for joining me in this bet against the odds. Be excellent to each other.
16:04:30.470 npm warn deprecated
16:04:30.471 npm warn deprecated (For a CapTP with native promises, see @endo/eventual-send and @endo/captp)
16:04:30.556 npm warn deprecated workbox-google-analytics@6.6.0: It is not compatible with newer versions of GA starting with v4, as long as you are using GAv3 it should be ok, but the package is not longer being maintained
16:04:31.698 npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
16:04:31.943 npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
16:04:32.679 npm warn deprecated domexception@2.0.1: Use your platform's native DOMException instead
16:04:33.491 npm warn deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
16:04:33.582 npm warn deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
16:04:34.281 npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
16:04:34.385 npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
16:04:34.920 npm warn deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
16:04:34.928 npm warn deprecated @babel/plugin-proposal-private-methods@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-methods instead.
16:04:34.942 npm warn deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
16:04:34.943 npm warn deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
16:04:34.955 npm warn deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
16:04:35.959 npm warn deprecated source-map@0.8.0-beta.0: The work that was done in this beta branch won't be included in future versions
16:04:36.182 npm warn deprecated @babel/plugin-proposal-private-property-in-object@7.21.11: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-property-in-object instead.
16:04:38.947 npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.
16:04:41.550 
16:04:41.551 added 1336 packages, and audited 1337 packages in 16s
16:04:41.552 
16:04:41.552 269 packages are looking for funding
16:04:41.552   run `npm fund` for details
16:04:41.560 
16:04:41.561 9 vulnerabilities (3 moderate, 6 high)
16:04:41.561 
16:04:41.561 To address all issues (including breaking changes), run:
16:04:41.561   npm audit fix --force
16:04:41.561 
16:04:41.561 Run `npm audit` for details.
16:04:41.565 npm notice
16:04:41.565 npm notice New minor version of npm available! 11.6.2 -> 11.7.0
16:04:41.565 npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.7.0
16:04:41.565 npm notice To update run: npm install -g npm@11.7.0
16:04:41.565 npm notice
16:04:42.490 Could not find a required file.
16:04:42.491   Name: index.js
16:04:42.491   Searched in: /vercel/path0/src
16:04:42.504 Error: Command "react-scripts build" exited with 1

**Reply by icyJoseph** [reply]
That's trying to build a create-react-app ~~ Not a Next.js application

**Reply by GabrielIdongesit** [reply]
Yes react-app

how do I fix it please

**Reply by icyJoseph** [reply]
Well, try to build it locally first. Does the project even start in dev mode? It says, could not find a required file: `index.js` 🤷
